includes/uninstall-functions.php

```diff
@@ -98,33 +84,25 @@ function delete_terms() {
 
 	global $wpdb;
 
-	$current_page = 0;
-	$per_page     = 1000;
-	$taxonomy     = 'amp_validation_error';
-
-	do {
-		$offset = $per_page * $current_page;
-
-		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Cannot cache result since we're deleting the records.
-		$term_ids = $wpdb->get_col(
-			$wpdb->prepare(
-				"SELECT term_id FROM $wpdb->term_taxonomy WHERE taxonomy = %s LIMIT %d OFFSET %d;",
-				$taxonomy,
-				$per_page,
-				$offset
-			)
-		);
-
-		if ( empty( $term_ids ) || ! is_array( $term_ids ) ) {
-			break;
-		}
+	// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
 
-		foreach ( $term_ids as $term_id ) {
-			wp_delete_term( $term_id, $taxonomy );
-		}
+	$wpdb->delete(
+		$wpdb->term_taxonomy,
+		[
+			'taxonomy' => 'amp_validation_error',
+		]
+	);
+
+	// Delete orphan relationships.
+	$wpdb->query( "DELETE tr FROM $wpdb->term_relationships AS tr LEFT JOIN $wpdb->posts AS posts ON posts.ID = tr.object_id WHERE posts.ID IS NULL;" );
+
+	// Delete orphan terms.
+	$wpdb->query( "DELETE t FROM $wpdb->terms AS t LEFT JOIN $wpdb->term_taxonomy AS tt ON t.term_id = tt.term_id WHERE tt.term_id IS NULL;" );
 
-		$current_page++;
-	} while ( ! empty( $result ) );
+	// Delete orphan term meta.
+	$wpdb->query( "DELETE tm FROM $wpdb->termmeta AS tm LEFT JOIN $wpdb->term_taxonomy AS tt ON tm.term_id = tt.term_id WHERE tt.term_id IS NULL;" );
```

Comment:
WooCommerce has a [check](https://github.com/woocommerce/woocommerce/blob/5239cf62e74ba9284578e73d97353dc1e10c8fa4/plugins/woocommerce/uninstall.php#L75-L76):

```php
	// Delete terms if > WP 4.2 (term splitting was added in 4.2).
	if ( version_compare( $wp_version, '4.2', '>=' ) ) {
```

It's not clear to me why this check as added. I guess it's to guard against accidentally deleting terms which are shared with other taxonomies? In our case here it is extremely unlikely that terms would be shared across taxonomies because the term slug is a MD5 hash of the object. So I don't think this is a concern.

That being said, I wonder if it would be safer to reverse the order here so that we don't rely on orphaned rows. 

For example, if we look at the SQL queries provided by @milindmore22 in https://wordpress.org/support/topic/how-to-delete-plugin-data-completely/#post-15091459

We can first delete term meta, then delete term relationships, then delete terms, and finally delete the taxonomy. This will ensure that we only ever delete items that are connected with the `amp_validation_error` taxonomy.

Reply:
Nevertheless, the queries are not as efficient as they rely on subqueries to work:

```sql
DELETE FROM wp_termmeta WHERE term_id 
  IN ( SELECT term_id FROM wp_term_taxonomy WHERE taxonomy = 'amp_validation_error' )
DELETE FROM wp_term_relationships WHERE term_taxonomy_id 
  IN ( SELECT term_taxonomy_id FROM wp_term_taxonomy WHERE taxonomy = 'amp_validation_error' )
DELETE FROM wp_terms WHERE term_id 
  IN (SELECT term_id FROM wp_term_taxonomy WHERE taxonomy = 'amp_validation_error' )
```

Reply:
Well, a `LEFT JOIN` is also not particularly efficient. In any case, the subqueries could be rewritten to use joins I believe. 

I _think_ these are the equivalent queries:

```sql
DELETE wp_termmeta
	FROM wp_termmeta tm
	INNER JOIN wp_term_taxonomy tt
	ON tt.term_id = tm.term_id
	WHERE tt.taxonomy = 'amp_validation_error';

DELETE wp_term_relationships
	FROM wp_term_relationships tr
	INNER JOIN wp_term_taxonomy tt
	ON tr.term_taxonomy_id = tt.term_taxonomy_id
	WHERE tt.taxonomy = 'amp_validation_error';

DELETE wp_terms
	FROM wp_terms t
	INNER JOIN wp_term_taxonomy tt
	ON t.term_id = tt.term_id
	WHERE taxonomy = 'amp_validation_error';
```

Which can then be followed up by:

```php
$wpdb->delete(
	$wpdb->term_taxonomy,
	[
		'taxonomy' => 'amp_validation_error',
	]
);
```